### DISCUSSION TITLE
How can I get the width and height or the x and y position of a widget?

### DISCUSSION THREAD

**Discussion by LaBatata101** [emilk/egui#1138] [Q&A] [answered]
didn't find anything in the docs, is this possible?

**Comment by eugineerd** [comment, marked as answer]
Use the [`rect`](https://docs.rs/egui/latest/egui/struct.Rect.html) field of a [`Response`](https://docs.rs/egui/latest/egui/struct.Response.html) returned from adding a widget to the ui.
```rust
let label_response = ui.label("Text");
let label_height = label_response.rect.height();
let label_pos = label_response.rect.left_top();
```

**Reply by LaBatata101** [reply]
Thanks!
